### PR TITLE
updated the dev installation guides

### PR DESCRIPTION
**Pull request by techfreaque**

### CODE REVIEW

File: installation/developer-installation/for-developers.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Developer installation
+# simple developer installation
```
Comment (GuillaumeDSM) [inline installation/developer-installation/for-developers.md]:
```suggestion
# Tentacle developer installation
```

File: installation/developer-installation/full-developer-installation.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+* ****[**create starting scripts in pycharm**](https://app.gitbook.com/s/XGhS0OJcF37WFUAtCJrB/c/K2S34tUZ8NdXf7pYP47C/installation/developer-installation/full-developer-installation#create-starting-scripts-in-pycharm)****
+* ****[**start OctoBot in pycharm**](https://app.gitbook.com/s/XGhS0OJcF37WFUAtCJrB/c/K2S34tUZ8NdXf7pYP47C/installation/developer-installation/full-developer-installation#start-octobot-in-pycharm)****
+
+## Install OctoBot requirements
+
+**Download and install:**
+
+* IDE: [PyCharm](https://www.jetbrains.com/pycharm/)
+* SCM: [Git](https://git-scm.com/downloads)
+*   we also use [GitKraken](https://www.gitkraken.com/git-client) to easily manage
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:24]:
```suggestion
*   We also use [GitKraken](https://www.gitkraken.com/git-client) to easily manage
```

File: installation/developer-installation/full-developer-installation.md [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+    OctoBot's multiple repos, this is just a quality of life
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:26]:
```suggestion
    OctoBot's multiple repos, this is just a quality of life improvement and is not necessary.
```
(no newline)

File: installation/developer-installation/full-developer-installation.md [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+    improvement and is not necessary.
+* Language: [Python 3.8](https://www.python.org/downloads/)
+
+#### Dependencies for Windows
+
+* download [Visual Studio build tools](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019) and install "Desktop development with C++"
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:33]:
```suggestion
* Download [Visual Studio build tools](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019) and install "Desktop development with C++"
```

File: installation/developer-installation/full-developer-installation.md [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+![](<../../.gitbook/assets/image (6).png>)
+
+![](<../../.gitbook/assets/image (1).png>)
+
+#### Dependencies for Mac
+
+* install [GCC](https://discussions.apple.com/thread/8336714)
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:41]:
```suggestion
* Install [GCC](https://discussions.apple.com/thread/8336714)
```

File: installation/developer-installation/full-developer-installation.md [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+### Install Octobot dependencies
+
+* For each OctoBot's repository: install missing dependencies in requirements.txt and dev\_requirements.txt.
+
+{% hint style="warning" %}
+Do not install the requirements related to the previously downloaded repositories or your python runner will use them instead of your local code version.
+{% endhint %}
+
+## create starting scripts in pycharm
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:203]:
```suggestion
## Create starting scripts in pycharm
```

File: installation/developer-installation/full-developer-installation.md [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+[![run configuration](https://raw.githubusercontent.com/Drakkar-Software/OctoBot/assets/wiki\_resources/run\_config.png)](https://raw.githubusercontent.com/Drakkar-Software/OctoBot/assets/wiki\_resources/run\_config.png)
+
+### OctoBot starting script
+
+{% hint style="info" %}
+this script will run OctoBot and apply all changes made to all repositories except the OctoBot-Tentacles folder
+{% endhint %}
+
+* click on Add Configuration
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:215]:
```suggestion
* Click on Add Configuration
```

File: installation/developer-installation/full-developer-installation.md [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+
+![](<../../.gitbook/assets/image (12).png>)
+
+now add a new python script
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:219]:
```suggestion
Add a new python script
```

File: installation/developer-installation/full-developer-installation.md [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+
+![](<../../.gitbook/assets/image (19).png>)
+
+### install OctoBot-Tentacles and run OctoBot script
+
+{% hint style="info" %}
+this script will install the OctoBot-Tentacles folder and run Octobot with all your changes applied from all repositories.
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:226]:
```suggestion
This script will install the OctoBot-Tentacles folder and run Octobot with all your changes applied from all repositories.
```

File: installation/developer-installation/full-developer-installation.md [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+
+* click on Add Configuration
+
+![](<../../.gitbook/assets/image (12).png>)
+
+now add a new python script
+
+![](<../../.gitbook/assets/image (19).png>)
+
+### install OctoBot-Tentacles and run OctoBot script
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:223]:
```suggestion
### Install OctoBot-Tentacles and run the OctoBot script
```

File: installation/developer-installation/full-developer-installation.md [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+
+{% hint style="info" %}
+this script will install the OctoBot-Tentacles folder and run Octobot with all your changes applied from all repositories.
+{% endhint %}
+
+#### create the following three scripts:
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:229]:
```suggestion
#### Create the following three scripts:
```

File: installation/developer-installation/full-developer-installation.md [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+
+* **Script to install OctoBot-Tentacles package**
+
+![](<../../.gitbook/assets/image (14).png>)
+
+* **Script to zip, install and run in one go**
+
+![](<../../.gitbook/assets/image (2).png>)
+
+## start OctoBot in pycharm
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:243]:
```suggestion
## Start OctoBot in pycharm
```

File: installation/developer-installation/full-developer-installation.md [hunk 13]
```diff
@@ -0,0 +1,247 @@
...
+
+* now you should have all the Octobot modules in one folder
+
+![](<../../.gitbook/assets/image (8).png>)
+
+#### OctoBot repositories
+
+OctoBot code is split in multiple repositories:
+
+*   [https://github.com/Drakkar-Software/OctoBot](https://github.com/Drakkar-Software/OctoBot) (dev branch) for the
```
Comment (GuillaumeDSM) [inline installation/developer-installation/full-developer-installation.md:120]: in this listing, we should remove all these newlines inside items, it creates weird line breaks

**Review by GuillaumeDSM — COMMENTED** [review]
amazing work, thanks a lot @techfreaque

File: SUMMARY.md [hunk 14]
```diff
@@ -5,20 +5,22 @@
...
+* [OctoBot architecture](guides/developer-guide.md)
 * [Contribution](guides/contribution.md)
 
 ## Coding
 
 * [Developer tips](coding/developers-tips.md)
 * [Tests](coding/tests.md)
 
-## Tentacles <a id="coding-1"></a>
+## Tentacles <a href="coding-1" id="coding-1"></a>
```
Comment (Herklos) [inline SUMMARY.md:24]: Is it working?

File: coding/tentacle-development.md [hunk 15]
```diff
@@ -72,4 +72,3 @@ class SuperSimpleStrategyEvaluator(SimpleStrategyEvaluator):
        # trading modes should wake up after this update
        await self.strategy_completed(cryptocurrency, symbol)
 ```
-
```
Comment (Herklos) [inline coding/tentacle-development.md:75]: ?

File: guides/developer-guide.md [hunk 16]
```diff
@@ -1,17 +1,9 @@
-# Developer startup guide
+# OctoBot architecture
 
-{% hint style="note" %}
+{% hint style="info" %}
 This page is referring to OctoBot in versions superior to 0.4.0.
```
Comment (Herklos) [inline guides/developer-guide.md:4]: I think we can remove it, there is no 0.3.X running anymore, what do you think @GuillaumeDSM?

File: installation/developer-installation/full-developer-installation.md [hunk 17]
```diff
@@ -0,0 +1,247 @@
...
+description: Setting up the recommended OctoBot development environment
+---
+
+# full developer installation
+
+{% hint style="info" %}
+This page is referring to OctoBot in versions superior to 0.4.0.
+{% endhint %}
+
+* ****[**Install OctoBot requirements**](https://app.gitbook.com/s/XGhS0OJcF37WFUAtCJrB/c/K2S34tUZ8NdXf7pYP47C/guides/dev-environment-setup#install-octobot-requirements)****
```
Comment (Herklos) [inline installation/developer-installation/full-developer-installation.md:11]: I don't think we should keep these links. You should have links using docs.octobot.online or just relative links

File: installation/developer-installation/full-developer-installation.md [hunk 18]
```diff
@@ -0,0 +1,247 @@
...
+    improvement and is not necessary.
+* Language: [Python 3.8](https://www.python.org/downloads/)
+
+#### Dependencies for Windows
+
+* download [Visual Studio build tools](https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2019) and install "Desktop development with C++"
+
+![](<../../.gitbook/assets/image (6).png>)
+
+![](<../../.gitbook/assets/image (1).png>)
```
Comment (Herklos) [inline installation/developer-installation/full-developer-installation.md:37]: Can you rename your image assets to have a name that matches the image content please?

**Review by Herklos — CHANGES_REQUESTED** [review]
Thanks a lot for improving OctoBot documentation!

**Pull request closed without merging**